### PR TITLE
Add missing values to placeholder metadata

### LINKED ISSUE #285: [BUG] template-analyzer throws TemplateValidationException for subscription().displayName for bicep as The language expression property 'displayName' doesn't exist
### Describe the bug

Below is the sample code:

```
param name string = 'vnet-${subscription().displayName}'
param location string
resource vnet 'Microsoft.Network/virtualNetworks@2022-01-01' = {
  name: name
  location: location
}
```

Gives error:

```
Error: An exception occurred while analyzing a template
Exception details:
Microsoft.Azure.Templates.Analyzer.Core.TemplateAnalyzerException: Error while processing template.
 ---> Azure.Deployments.Templates.Exceptions.TemplateValidationException: The template resource 'name' at line '12' and column '13' is not valid: The language expression property 'displayName' doesn't exist, available properties are 'id, subscriptionId, tenantId'.. Please see https://aka.ms/arm-template-expressions for usage details.
 ---> Azure.Deployments.Core.Exceptions.ExpressionException: The language expression property 'displayName' doesn't exist, available properties are 'id, subscriptionId, tenantId'.
   at Azure.Deployments.Expression.Expressions.FunctionExpression.SelectObjectProperty(JObject root, JToken token, TemplateErrorAdditionalInfo additionalInfo)
   at Azure.Deployments.Expression.Expressions.FunctionExpression.SelectProperty(JToken root, JToken property, TemplateErrorAdditionalInfo additionalInfo)
   at Azure.Deployments.Expression.Expressions.FunctionExpression.EvaluateExpression(ExpressionEvaluationContext context, TemplateErrorAdditionalInfo additionalInfo)
   at Azure.Deployments.Expression.Expressions.FunctionExpression.<>c__DisplayClass23_0.<EvaluateParameters>b__0(LanguageExpression parameter)
   at System.Array.ConvertAll[TInput,TOutput](TInput[] array, Converter`2 converter)
   at Microsoft.WindowsAzure.ResourceStack.Common.Extensions.IEnumerableExtensions.SelectArray[TSource,TElement](TSource[] source, Converter`2 elementSelector)
   at Azure.Deployments.Expression.Expressions.FunctionExpression.EvaluateParameters(ExpressionEvaluationContext context, TemplateErrorAdditionalInfo additionalInfo)
   at Azure.Deployments.Expression.Expressions.FunctionExpression.EvaluateExpression(ExpressionEvaluationContext context, TemplateErrorAdditionalInfo additionalInfo)
   at Azure.Deployments.Expression.Engines.ExpressionsEngine.EvaluateLanguageExpressionsInternal(JToken root, ExpressionEvaluationContext evaluationContext, Boolean ignoreExceptions, TemplateErrorAdditionalInfo additionalInfo, InsensitiveHashSet skipEvaluationPaths)
   at Azure.Deployments.Expression.Engines.ExpressionsEngine.EvaluateLanguageExpressionsRecursive(JToken root, ExpressionEvaluationContext evaluationContext, TemplateErrorAdditionalInfo additionalInfo, InsensitiveHashSet skipEvaluationPaths)
   at Azure.Deployments.Templates.Engines.TemplateParameterizationEngine.ParameterizeInputParameter(String parameterName, TemplateInputParameter parameter, InsensitiveDictionary`1 inputParameters, ExpressionEvaluationContext evaluationContext, Boolean ignoreEvaluationExceptionsForWhatIf)
   at Azure.Deployments.Templates.Engines.TemplateParameterizationEngine.ParameterizeInputParameterRecursive(String parameterName, TemplateInputParameter parameter, InsensitiveDictionary`1 inputParameters, InsensitiveDictionary`1 recursionState, ExpressionEvaluationContext evaluationContext, Boolean ignoreEvaluationExceptionsForWhatIf)
   --- End of inner exception stack trace ---
   at Azure.Deployments.Templates.Engines.TemplateParameterizationEngine.ParameterizeInputParameterRecursive(String parameterName, TemplateInputParameter parameter, InsensitiveDictionary`1 inputParameters, InsensitiveDictionary`1 recursionState, ExpressionEvaluationContext evaluationContext, Boolean ignoreEvaluationExceptionsForWhatIf)
   at Azure.Deployments.Templates.Engines.TemplateParameterizationEngine.ParameterizeInputParameters(Template template, InsensitiveDictionary`1 inputParameters, InsensitiveDictionary`1 metadata, Boolean ignoreEvaluationExceptionsForWhatIf, InsensitiveDictionary`1 functionEvaluationOverwrites)
   at Azure.Deployments.Templates.Engines.TemplateParameterizationEngine.ParameterizeTemplate(Template template, InsensitiveDictionary`1 inputParameters, InsensitiveDictionary`1 metadata, Boolean ignoreEvaluationExceptionsForWhatIf, InsensitiveDictionary`1 functionEvaluationOverwrites)
   at Microsoft.Azure.Templates.Analyzer.TemplateProcessor.ArmTemplateProcessor.ParseAndValidateTemplate(InsensitiveDictionary`1 parameters, InsensitiveDictionary`1 metadata)
   at Microsoft.Azure.Templates.Analyzer.Core.TemplateAnalyzer.AnalyzeTemplate(String template, String templateFilePath, String parameters)
   --- End of inner exception stack trace ---
   at Microsoft.Azure.Templates.Analyzer.Core.TemplateAnalyzer.AnalyzeTemplate(String template, String templateFilePath, String parameters)
   at Microsoft.Azure.Templates.Analyzer.Cli.CommandLineParser.AnalyzeTemplate(FileInfo templateFilePath, FileInfo parametersFilePath)

Execution summary:
        The verbose mode (option -v or --verbose) can be used to obtain even more information about the execution.

        Summary of the errors:
                1 instance of: An exception occurred while analyzing a template

        0 Warnings
        1 Error
```

### Expected behavior

Should not throw any error. 

### Reproduction Steps

Copy below code to **demo.bicep**
```
param name string = 'vnet-${subscription().displayName}'
param location string
resource vnet 'Microsoft.Network/virtualNetworks@2022-01-01' = {
  name: name
  location: location
}
```

now run command TemplateAnalyzer to analyze demo.bicep

`TemplateAnalyzer.exe analyze-template demo.bicep`

### Environment

`TemplateAnalyzer.exe --version`
  0.3.0

`dotnet --version`
  6.0.202

vscode (IDE): 1.71.0

### PR DESCRIPTION
**Pull request by VeraBE**
## Description

Closes #285

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [x] The pull request does not introduce breaking changes.

### [General Guidelines](../CONTRIBUTING.md)
- [x] Title of the pull request is clear and informative.
- [x] Description of the pull request is clear and informative.
- [x] I have added myself to the 'assignees'.
- [x] I have added 'linked issues' if relevant.

### [Testing Guidelines](../CONTRIBUTING.md#tests)
- [] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/Analyzer.TemplateProcessor/PlaceholderInputGenerator.cs [hunk 1]
```diff
@@ -186,7 +186,8 @@ internal static InsensitiveDictionary<JToken> GeneratePlaceholderDeploymentMetad
                 { "subscription", new JObject(
                     new JProperty("id", "/subscriptions/00000000-0000-0000-0000-000000000000"),
                     new JProperty("subscriptionId", "00000000-0000-0000-0000-000000000000"),
-                    new JProperty("tenantId", "00000000-0000-0000-0000-000000000000")) },
+                    new JProperty("tenantId", "00000000-0000-0000-0000-000000000000"),
+                    new JProperty("displayName", "aSubscriptionName")) },
```
Comment (larspmohr) [inline src/Analyzer.TemplateProcessor/PlaceholderInputGenerator.cs:190]: nit: maybe just 'SubscriptionName'

**Review by larspmohr — APPROVED** [review]

File: src/Analyzer.TemplateProcessor/PlaceholderInputGenerator.cs [hunk 2]
```diff
@@ -186,7 +186,8 @@ internal static InsensitiveDictionary<JToken> GeneratePlaceholderDeploymentMetad
...
                 { "resourceGroup", new JObject(
```
Comment (JohnathonMohr) [inline src/Analyzer.TemplateProcessor/PlaceholderInputGenerator.cs:191]: There are several other metadata fields supported in templates, but not initialized here. Can we fill out this section completely to get ahead of other parsing failures later?

File: src/Analyzer.TemplateProcessor/PlaceholderInputGenerator.cs (hunk 1 above)
Comment (VeraBE) [reply src/Analyzer.TemplateProcessor/PlaceholderInputGenerator.cs:190]: I would rather leave it as it is to make it more clear that is just one possible subscription name

File: src/Analyzer.TemplateProcessor/PlaceholderInputGenerator.cs [hunk 3]
```diff
@@ -160,40 +160,62 @@ internal static InsensitiveDictionary<JToken> GeneratePlaceholderDeploymentMetad
...
             {
-                ""id"": ""/providers/Microsoft.Management/managementGroups/examplemg1"",
-                ""name"": ""examplemg1"",
+                ""id"": ""/providers/Microsoft.Management/managementGroups/placeholderManagementGroup"",
+                ""name"": ""placeholderManagementGroup"",
                 ""properties"": {
                   ""details"": {
                     ""parent"": {
-                      ""displayName"": ""Tenant Root Group"",
+                      ""displayName"": ""Placeholder Tenant"",
```
Comment (JohnathonMohr) [inline src/Analyzer.TemplateProcessor/PlaceholderInputGenerator.cs:170]: How about "Placeholder Tenant Root Group"?  It's not actually a tenant.

**Review by JohnathonMohr — APPROVED** [review]

**Pull request merged**